cfunits/units.py

```diff
@@ -1304,6 +1309,49 @@ def _new_reason_notvalid(self, reason):
     # ----------------------------------------------------------------
     # Attributes
     # ----------------------------------------------------------------
+    @property
+    def has_offset(self):
+        '''True if the units contain an offset.
+        
+    Note that if a multiplicative component of the units had an offset
+    during instantiation, then the offset is ignored in the resulting
+    `Units` object. See below for examples.
+
+    **Examples**
+
+    >>> Units('K').has_offset
+    False
+    >>> Units('K @ 0').has_offset
+    False
+    >>> Units('K @ 273.15').has_offset
+    True
+    >>> Units('degC').has_offset
+    True
+    >>> Units('degF').has_offset
+    True
+
+    >>> Units('Watt').has_offset
+    False
+    >>> Units('m2.kg.s-3').has_offset
+    False
+
+    >>> Units('km').has_offset
+    False
+    >>> Units('1000 m').has_offset
+    False
+        
+    >>> Units('m2.kg.s-3 @ 3.14').has_offset
+    True
+    >>> Units('(K @ 273.15) m s-1').has_offset
+    False
+    >>> Units('degC m s-1').has_offset
+    False
+    >>> Units('degC m s-1') == Units('K m s-1')
+    True
+
+        '''
+        return '@' in self.formatted()
```

Comment:
So simple yet works flawlessly (at least excluding potential horrible edge cases or complications but I've tried and can't thin or find any of those)! Great idea.

Looking through the example, my only concern was that ``@ 0`` may imply an offset, but the standardisation by the ``formatted`` method manages all of that, as implied by the unit test pass for ``self.assertFalse(Units('K @ 0').has_offset)``:

```python
>>> cfunits.Units('K @ 0').formatted()
'K'
```

Reply:
> at least excluding potential horrible edge cases or complications but I've tried and can't thin or find any of those

Sadly it seems I may have found one (relating to datetimes, naturally), see general review comment below. :disappointed: